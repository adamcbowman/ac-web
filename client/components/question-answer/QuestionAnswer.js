import React from 'react'
import PropTypes from 'prop-types'
import { InnerHTML } from '~/components/misc'
import Question from './Question'
import Answer from './Answer'

QuestionAnswer.propTypes = {
    question: PropTypes.string.isRequired,
    answer: PropTypes.string.isRequired,
}

export default function QuestionAnswer({ question, answer }) {
    return (
        <section>
            <Question>{question}</Question>
            <Answer>
                <InnerHTML>
                    {answer}
                </InnerHTML>
            </Answer>
        </section>
    )
}
