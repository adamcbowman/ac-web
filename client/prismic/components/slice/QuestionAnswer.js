import React from 'react'
import Base from 'components/question-answer'
import { StructuredText } from 'prismic/components/base'

export default function QuestionAnswer({ value }) {
    const [{ question, answer }] = value

    return (
        <Base question={question} answer={<StructuredText value={answer} />} />
    )
}
