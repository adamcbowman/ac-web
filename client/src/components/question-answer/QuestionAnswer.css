@import "../../styles/colors.css";

.Child {
    &:before {
        float: left;
        padding-right: 0.5em;
        text-transform: uppercase;
        font-weight: 700;
    }
}

.Question {
    composes: Child;

    font-weight: 700;

    &:before {
        content: 'Q:';
        color: var(--brand-secondary);
    }
}

.Answer {
    composes: Child;

    &:before {
        content: 'A:';
        color: var(--brand-primary);
    }
}
