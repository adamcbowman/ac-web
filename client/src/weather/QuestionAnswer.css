.Question {
    &:before {
        content: 'Question: ';
        font-weight: bold;
    }
}

.Answer {
    &:before {
        content: 'Answer: ';
        font-weight: bold;
    }
}
