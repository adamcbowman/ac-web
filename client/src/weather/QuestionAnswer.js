import React, { PropTypes } from 'react'
import CSSModules from 'react-css-modules'
import styles from './QuestionAnswer.css'

// TODO: Should be base component

QuestionAnswer.propTypes = {
    question: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
    answer: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
}

function QuestionAnswer({ question, answer }) {
    return (
        <section>
            <div styleName='Question'>{question}</div>
            <div styleName='Answer'>{answer}</div>
        </section>
    )
}

export default CSSModules(QuestionAnswer, styles)
