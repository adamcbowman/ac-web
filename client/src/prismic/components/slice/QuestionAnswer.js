import {compose, flattenProp, mapProps} from 'recompose'
import QuestionAnswer from 'components/question-answer'

export default compose(
    mapProps(props => props.content[0]),
    flattenProp('content'),
)(QuestionAnswer)
