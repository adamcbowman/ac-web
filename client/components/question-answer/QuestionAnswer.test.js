import React from 'react'
import Renderer from 'react-test-renderer'
import QuestionAnswer from './QuestionAnswer'

test('question & answer component', () => {
    const qa = Renderer.create(
        <QuestionAnswer question="A question" answer="An answer" />
    )

    expect(qa).toMatchSnapshot()
})
