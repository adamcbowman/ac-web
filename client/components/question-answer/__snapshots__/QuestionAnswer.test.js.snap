// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`question & answer component 1`] = `
<details
  className="Container"
  open={false}
>
  <summary>
    A question
  </summary>
  <div
    className="horizontal"
  >
    An answer
  </div>
</details>
`;
